src/lib389/lib389/utils.py

```diff
@@ -1460,3 +1461,86 @@ def is_fips():
+    """Get the status of a Task, if "watching" keep looping over the tasks
+    :param inst: DirSrv Object
+    :type inst: Object
+    :param log: Logging Object
+    :type log: Object
+    :param taskObj: Lib389 Object (task object from plugins.py)
+    :type taskObj: Object
+    :param dn: A task DN to get the status for (if omitted all tasks for checked)
+    :type dn: str
+    :param show_log: Show the task's log attribute (could be lengthy)
+    :type show_log: boolean
+    :param watch: Keep looping over the task until it completes
+    :type watch: boolean
+    :param use_json: Report the status in JSON
+    :type use_json: boolean
+    """
+
+    tasks = taskObj(inst).list()
+    all_finished = False
+    while not all_finished:
+        task_list = []
+        all_finished = True
+        for task in tasks:
+            task_base_dn = task.get_attr_val_utf8_l('basedn')
+            if task_base_dn is None:
+                # Try linked attr, it uses a different attr name
+                task_base_dn = task.get_attr_val_utf8_l('linkdn')
+            task_filter = task.get_attr_val_utf8_l('filter')
+            status = task.get_attr_val_utf8('nsTaskStatus')
+            exitcode = task.get_attr_val_utf8('nsTaskExitcode')
+            task_log =  task.get_attr_val_utf8('nsTaskLog')
+            if task_base_dn is None:
+                task_base_dn = ""
+            if task_filter is None:
+                task_filter = ""
+            if status is None:
+                status = ""
+            if exitcode is None:
+                all_finished = False
+                exitcode = "Not finished ..."
+
+            if not show_log:
+                task_log = ""
+
+            if (dn is not None and dn.lower() == task.dn.lower()) or dn is None:
+                    task_list.append({
+                        'dn': task.dn,
+                        'basedn': task_base_dn,
+                        'filter': task_filter,
+                        'log': task_log,
+                        'status': status,
+                        'exitcode': exitcode
+                    })
+
+        # Display the status
+        if use_json:
+            log.info(json.dumps({"type": "list", "items": task_list}, indent=4))
+        else:
+            num_of_tasks = len(task_list)
+            count = 1
+            plural = ""
+            if num_of_tasks != 1:
+                plural = "s"
+            log.info(f'Found {num_of_tasks} fix-up task{plural}\n')
+
+            for task in task_list:
+                log.info(f"[{count}] Task: {task['dn']}")
+                log.info("-" * 80)
+                log.info(f" - Base DN:   {task['basedn']}")
+                if task['filter']:
+                    log.info(f" - Filter:    {task['filter']}")
+                log.info(f" - Status:    {task['status']}")
+                log.info(f" - Exit Code: {task['exitcode']}")
+                if show_log:
+                    log.info(f" - Log:\n{task['log']}")
```

Comment:
I wonder if we should add a timestamp here. But it's more of a suggestion. :)

Reply:
For my part I would rather log the time elapsed since task startup ( should be in nsTaskCreated attribute) 
rather than a time stamp based on current time.

Reply:
I meant that we can log the time which is aligned with the error log time events. So the admin will see if the operation hanged/delayed on the exact stage.

But the overall "time elapsed" also sounds good!